### PR TITLE
Oldwayseed

### PR DESCRIPTION
**Pull request by jmezalon**

### CODE REVIEW

File: backend/db/queries/orderQueries.js [hunk 1]
```diff
@@ -32,7 +32,71 @@ const deleteOrder = (req, res, next) => {
...
+      res.status(200).json({
+        status: "success",
+        order,
+        message: "single order for user"
+      });
+    })
+    .catch(err => next(err));
+};
+
+const orderSummary = (req, res, next) => {
```
Comment (chrisdl) [inline backend/db/queries/orderQueries.js]: I've realized now a way to explain better why assigning res.status in the database layer (DAO) is a bad idea.

1. imagine you have two common queries and you want to perform them side by side instead of merging them into a single query through some `JOIN` black magic.
2.Imagine also you have a route which needs to perform a query AND perform an unrelated network request (whose response you need for your response). If the query sets the `res` response object you are toast. If instead the query just returns the results of the query you can perform the network request asynchronously and once both have happened you would use what parts you need of the both of them in the response.

Basically this makes it impossible for you to reuse the query functions, they MUST be the only thing a route calls, making them tightly coupled to the routes. This is why a separation of concerns is good, because it gives you the options and capability for reuse.
